### ISSUE TITLE
Monotone constraints do not work with 'RMSEWithUncertainty' loss

### ISSUE THREAD

**Issue by konderal333** [catboost/catboost#2780]
Problem: 
I am trying to create a model using monotone constraints and 'RMSEWithUncertainty' loss. However, when I examined the output for some manually crafted inputs, the monotone constraints for certain variables did not work. An example:

```python
model.get_params()

"""
{'iterations': 285,
 'learning_rate': 0.04690372882018757,
 'depth': 4,
 'l2_leaf_reg': 5.9175344238081,
 'loss_function': 'RMSEWithUncertainty',
 'silent': True,
 'random_strength': 4.530073899665095,
 'bagging_temperature': 0.11927102098609617,
'monotone_constraints': [1, 1, 1, 1, 1, 1, 1, 1, 0]}
"""

print(model.predict([80,40,80,80,80,80,50,3,1])[0])
print(model.predict([80,50,80,80,80,80,50,3,1])[0])

"""
3.8297171163569463
3.8285043184782883
"""
```

catboost version: 1.2.5
Operating System: Windows 11
CPU: Intel Core i7-1365U
GPU:  Intel UHD graphics
